src/Microsoft.AspNet.OData.Shared/Query/DefaultSkipTokenHandler.cs

```diff
@@ -0,0 +1,382 @@
+        /// </summary>
+        /// <param name="lastMember"> Object based on which SkipToken value will be generated.</param>
+        /// <param name="model">The edm model.</param>
+        /// <param name="orderByNodes">QueryOption </param>
+        /// <returns></returns>
+        protected virtual string GenerateSkipTokenValue(Object lastMember, IEdmModel model, IList<OrderByNode> orderByNodes)
+        {
+            object value;
+            if (lastMember == null)
+            {
+                return String.Empty;
+            }
+            IEnumerable<IEdmProperty> propertiesForSkipToken = GetPropertiesForSkipToken(lastMember, model, orderByNodes);
+
+            String skipTokenvalue = String.Empty;
+            if (propertiesForSkipToken == null)
+            {
+                return skipTokenvalue;
+            }
+
+            int count = 0;
+            int lastIndex = propertiesForSkipToken.Count() - 1;
+            foreach (IEdmProperty property in propertiesForSkipToken)
+            {
+                bool islast = count == lastIndex;
+                IEdmStructuredObject obj = lastMember as IEdmStructuredObject;
+                if (obj != null)
+                {
+                    obj.TryGetPropertyValue(property.Name, out value);
+                }
+                else
+                {
+                    value = lastMember.GetType().GetProperty(property.Name).GetValue(lastMember);
+                }
+
+                String uriLiteral = String.Empty;
+                if (value == null)
+                {
+                    uriLiteral = ODataUriUtils.ConvertToUriLiteral(value, ODataVersion.V401);
+                }
+                else if (TypeHelper.IsEnum(value.GetType()))
+                {
+                    ODataEnumValue enumValue = new ODataEnumValue(value.ToString(), value.GetType().FullName);
+                    uriLiteral = ODataUriUtils.ConvertToUriLiteral(enumValue, ODataVersion.V401, model);
+                    uriLiteral = "'enumType'" + uriLiteral;
+                }
+                else
+                {
+                    uriLiteral = ODataUriUtils.ConvertToUriLiteral(value, ODataVersion.V401, model);
+                }
+                skipTokenvalue += property.Name + propertyDelimiter + uriLiteral + (islast ? String.Empty : CommaDelimiter.ToString());
+                count++;
+            }
+            return skipTokenvalue;
+        }
+
+        /// <summary>
+        /// Apply the $skiptoken query to the given IQueryable.
+        /// </summary>
+        /// <param name="query">The original <see cref="IQueryable"/>.</param>
+        /// <param name="skipTokenQueryOption">The skiptoken query option which needs to be applied to this query option.</param>
+        /// <returns>The new <see cref="IQueryable"/> after the skiptoken query has been applied to.</returns>
+        public override IQueryable<T> ApplyTo<T>(IQueryable<T> query, SkipTokenQueryOption skipTokenQueryOption)
```

Comment:
>public override IQueryable<T> ApplyTo<T>(IQueryable<T> query, SkipTokenQueryOption skipTokenQueryOption) [](start = 8, length = 104)

why can't use the method signature same as SkipTokenQueryOption?


public IQueryable<T> ApplyTo<T>(IQueryable<T> query, ODataQuerySettings querySettings, IList<OrderByNode> orderByNodes)

Reply:
As per Mike's feedback, the base class for the skiptoken handler should have been independent of the default implementation. This signature allows for more powerful customizations for the next link. 

---
In reply to: [281401734](https://github.com/OData/WebApi/pull/1724#discussion_r281401734) [](ancestors = 281401734)